content/artifacts/highlight-active-artifact-planets/plugin.js

```diff
@@ -0,0 +1,173 @@
+    this.showArifactTypes = [7];
+  }
+
+
+  async render(container) {
+    container.style.minHeight = 'unset';
+    container.style.width = '300px';
+
+    let groupNameLabel = document.createElement('div');
+    groupNameLabel.innerHTML = 'Highlight all active planet or mine?'
+    container.appendChild(groupNameLabel);
+
+    let radioDiv = document.createElement('div');
+    let groupNames = ["All", "My"];
+    for (let groupName of groupNames) {
+      let radioInput = document.createElement('input');
+      radioInput.style.width = '10%';
+      radioInput.name = "groupradio";
+      radioInput.type = "radio";
+      radioInput.value = groupName;
+      if (groupName == this.showGroup) {
+        radioInput.checked = true;
+      }
+      radioInput.onchange = (evt) => {
+        if (evt.target.checked) {
+          this.showGroup = evt.target.value;
+        }
+      };
+      let radioLabel = document.createElement('label');
+      radioLabel.style.width = '30%';
+      radioLabel.innerHTML = groupName;
+      radioDiv.appendChild(radioInput);
+      radioDiv.appendChild(radioLabel);
+    }
+    container.appendChild(radioDiv);
+
+
+    let artifactTypeLabel = document.createElement('div');
+    artifactTypeLabel.innerHTML = 'ArtifactTypes to highlight.'
+    artifactTypeLabel.style.marginTop = '10px';
+    artifactTypeLabel.style.marginBottom = '5px';
+    container.appendChild(artifactTypeLabel);
+
+    for (let artifactType of ARTIFACT_TYPES) {
+      let checkboxDiv = document.createElement('div');
+      let checkboxInput = document.createElement('input');
+      checkboxInput.style.width = '10%';
+      checkboxInput.type = "checkbox";
+      checkboxInput.value = artifactType.value;
+      if (this.showArifactTypes.indexOf(artifactType.value) != -1) {
+        checkboxInput.checked = true;
+      }
+      checkboxInput.onchange = (evt) => {
+        if (evt.target.checked) {
+          this.showArifactTypes.push(parseInt(evt.target.value));
+        } else {
+          let index = this.showArifactTypes.indexOf(parseInt(evt.target.value));
+          if (index != -1) {
+            this.showArifactTypes.splice(index, 1);
+          }
+          this.showArifactTypes.sort();
+        }
+        //this.message.innerHTML = this.showArifactTypes;
+      }
+
+      let checkboxLabel = document.createElement('label');
+      checkboxLabel.style.width = '80%';
+      checkboxLabel.innerHTML = artifactType.text;
+      checkboxDiv.appendChild(checkboxInput);
+      checkboxDiv.appendChild(checkboxLabel);
+      container.appendChild(checkboxDiv);
+    }
+
+    container.appendChild(this.message);
+  }
+
+
+  draw(ctx) {
+    let planets = [];
+    if (this.showGroup == 'My') {
+      planets = df.getMyPlanets();
+    } else if (this.showGroup == 'All') {
+      planets = df.getAllPlanets();
+    }
+    for (let planet of planets) {
+      let artifact = df.getActiveArtifact(planet);
+      if (!artifact) {
+        continue;
+      }
```

Comment:
this same check also happens in the `drawArc` function,
i personally would remove this check from the `drawArc` function and would move this check:
```
  if (artifactTypes.indexOf(artifact.artifactType ) == -1)  {
    return false;
  }
```
from the `drawArc` function to here, then replace the `artifactTypes` argument with an `artfiact` argument and pass that to the `drawArc` function instead
this way you only have to call `df.getActiveArtifact(planet)` once per planet

Reply:
Only highlight the artifact you choose at the page，I have move the check outside of the function "drawArc".